diff --git a/openr/platform/LinuxPlatformMain.cpp b/openr/platform/LinuxPlatformMain.cpp
index 97ea1fb46..0cf4c1d01 100644
--- a/openr/platform/LinuxPlatformMain.cpp
+++ b/openr/platform/LinuxPlatformMain.cpp
@@ -29,9 +29,6 @@ main(int argc, char** argv) {

   folly::EventBase mainEvb;
   openr::EventBaseStopSignalHandler handler(&mainEvb);
-  handler.registerSignalHandler(SIGINT);
-  handler.registerSignalHandler(SIGQUIT);
-  handler.registerSignalHandler(SIGTERM);

   std::vector<std::thread> allThreads{};
