diff -ruN a/vpp/app/version.c b/vpp/app/version.c
--- a/vpp/app/version.c	2022-11-11 12:17:52.665882031 +0530
+++ b/vpp/app/version.c	2022-11-12 18:52:32.972360041 +0530
@@ -16,6 +16,8 @@
 #include <vppinfra/cpu.h>
 #include <vpp/app/version.h>
 
+#define VPP_DISPLAY_VER "21.01"
+
 /** \file
     Display image version information
 */
@@ -29,7 +31,7 @@
 
 /** The image version string */
 char *vpe_version_string =
-  "vpp v" VPP_BUILD_VER
+  "vpp v" VPP_DISPLAY_VER " ("VPP_BUILD_VER")"
   " built by " VPP_BUILD_USER " on " VPP_BUILD_HOST " at " VPP_BUILD_DATE;
 
 /** The name of the compiler */
@@ -73,8 +75,8 @@
 
   if (verbose)
     {
+      vlib_cli_output (vm, "%-25s %s (%s)\n", "Version:", "v"VPP_DISPLAY_VER, VPP_BUILD_VER);
 #define _(a,b,c) vlib_cli_output (vm, "%-25s " b, a ":", c);
-      _("Version", "%s", "v" VPP_BUILD_VER);
       _("Compiled by", "%s", VPP_BUILD_USER);
       _("Compile host", "%s", VPP_BUILD_HOST);
       _("Compile date", "%s", VPP_BUILD_DATE);
