From 06f1e141bf7ed802d04ba9df90642bb0733319ec Mon Sep 17 00:00:00 2001
From: Lahari-Kamasetty <k.lahari@capgemini.com>
Date: Tue, 15 Nov 2022 12:33:11 +0530
Subject: [PATCH] VPP displayed version string improvements

Signed-off-by: Lahari-Kamasetty <k.lahari@capgemini.com>
---
 src/vpp/app/version.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/vpp/app/version.c b/src/vpp/app/version.c
index db13c8639..ef048ca0a 100644
--- a/src/vpp/app/version.c
+++ b/src/vpp/app/version.c
@@ -16,6 +16,8 @@
 #include <vppinfra/cpu.h>
 #include <vpp/app/version.h>
 
+#define VPP_DISPLAY_VER "21.01"
+
 /** \file
     Display image version information
 */
@@ -29,7 +31,7 @@
 
 /** The image version string */
 char *vpe_version_string =
-  "vpp v" VPP_BUILD_VER
+  "vpp v" VPP_DISPLAY_VER " (v"VPP_BUILD_VER")"
   " built by " VPP_BUILD_USER " on " VPP_BUILD_HOST " at " VPP_BUILD_DATE;
 
 /** The name of the compiler */
@@ -74,7 +76,7 @@ show_vpe_version_command_fn (vlib_main_t * vm,
   if (verbose)
     {
 #define _(a,b,c) vlib_cli_output (vm, "%-25s " b, a ":", c);
-      _("Version", "%s", "v" VPP_BUILD_VER);
+      _("Version", "%s", "v" VPP_DISPLAY_VER " (v"VPP_BUILD_VER")");
       _("Compiled by", "%s", VPP_BUILD_USER);
       _("Compile host", "%s", VPP_BUILD_HOST);
       _("Compile date", "%s", VPP_BUILD_DATE);
-- 
2.25.1

