commit d0af7b1b4b02c1d8d910f8efcfe82d21d7216747
Author: Thomas Mangin <thomas.mangin@exa.net.uk>
Date:   Sun Mar 15 21:43:02 2020 +0000

    fixes #962

diff --git a/lib/exabgp/reactor/loop.py b/lib/exabgp/reactor/loop.py
index c62d14d1..a13f8def 100644
--- a/lib/exabgp/reactor/loop.py
+++ b/lib/exabgp/reactor/loop.py
@@ -293,6 +293,7 @@ class Reactor (object):
 		workers = {}
 		peers = set()
 		api_fds = []
+		ms_sleep = int(self._sleep_time * 1000)
 
 		while True:
 			try:
@@ -333,7 +334,7 @@ class Reactor (object):
 				if self._completed(peers):
 					reload_completed = True
 
-				sleep = self._sleep_time
+				sleep = ms_sleep
 
 				# do not attempt to listen on closed sockets even if the peer is still here
 				for io in list(workers.keys()):
