diff --git a/proxygen/httpclient/CMakeLists.txt b/proxygen/httpclient/CMakeLists.txt
index b865f2a8..76438864 100644
--- a/proxygen/httpclient/CMakeLists.txt
+++ b/proxygen/httpclient/CMakeLists.txt
@@ -4,4 +4,6 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-add_subdirectory(samples)
+if(BUILD_TESTS)
+  add_subdirectory(samples)
+endif()
diff --git a/proxygen/httpserver/CMakeLists.txt b/proxygen/httpserver/CMakeLists.txt
index 40046e55..c79f5067 100644
--- a/proxygen/httpserver/CMakeLists.txt
+++ b/proxygen/httpserver/CMakeLists.txt
@@ -27,10 +27,12 @@ install(
     DESTINATION lib
 )
 
+if(BUILD_TESTS)
 add_executable(proxygen_push
     samples/push/PushServer.cpp
     samples/push/PushRequestHandler.cpp
 )
+
 target_compile_options(
     proxygen_push
     PRIVATE
@@ -144,6 +146,8 @@ if (BUILD_QUIC)
   )
 endif()
 
+endif() # if(BUILD_TESTS)
+
 file(
     GLOB_RECURSE PROXYGEN_HTTPSERVER_HEADERS_TOINSTALL
     RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
