diff --git a/configure.ac b/configure.ac
index 88c3b11..2292466 100644
--- a/configure.ac
+++ b/configure.ac
@@ -68,8 +68,8 @@ m4_warn([obsolete],
   ISO C90 headers unconditionally.])dnl
 # Autoupdate added the next two lines to ensure that your configure
 # script's behavior did not change.  They are probably safe to remove.
-AC_CHECK_INCLUDES_DEFAULT
-AC_PROG_EGREP
+#AC_CHECK_INCLUDES_DEFAULT
+#AC_PROG_EGREP
 
 
 # Check for systems which need -lsocket and -lnsl
